shuup/admin/modules/orders/object_selector.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2021, Shuup Commerce Inc. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Iterable, Tuple
+
+from shuup.admin.utils.permissions import has_permission
+from shuup.admin.views.select import BaseAdminObjectSelector
+from shuup.campaigns.models.campaigns import Coupon
+from shuup.discounts.models import Discount
+
+
+class CouponAdminObjectSelector(BaseAdminObjectSelector):
+    ordering = 7
+
+    @classmethod
+    def handles_selector(cls, selector):
+        return selector == "shuup.coupon"
+
+    def has_permission(self, user):
+        return has_permission(user, "coupon.object_selector")
+
+    def get_objects(self, search_term, *args, **kwargs) -> Iterable[Tuple[int, str]]:
+        """
+        Returns an iterable of tuples of (id, text)
+        """
+        objects = list(
```

Comment:
Any business logic to add here? filtering by shop?

Reply:
for coupons yes. you need to move this to `shuup.campaigns` app first (shuup/campaigns/admin_module).

- if there is a supplier (self.supplier) use that as a filter
- use the self.shop as a filter
- only active coupons

Reply:
Perfect